File: miniprogram/convert_svg_to_png.py
import os
import subprocess
import sys

def convert_svg_to_png(svg_file, png_file, size=81):
    try:
        from cairosvg import svg2png
        with open(svg_file, 'rb') as f:
            svg_data = f.read()
        svg2png(bytestring=svg_data, write_to=png_file, output_width=size, output_height=size)
        print(f"成功转换: {svg_file} -> {png_file}")
        return True
    except ImportError:
        print("cairosvg库未安装，尝试使用其他方法...")
        return False
    except Exception as e:
        print(f"转换失败: {svg_file} - {str(e)}")
        return False

def convert_svg_to_png_pillow(svg_file, png_file, size=81):
    try:
        from PIL import Image
        import cairosvg
        
        svg_data = open(svg_file, 'rb').read()
        png_data = cairosvg.svg2png(bytestring=svg_data, output_width=size, output_height=size)
        
        img = Image.open(io.BytesIO(png_data))
        img.save(png_file, 'PNG')
        print(f"成功转换: {svg_file} -> {png_file}")
        return True
    except ImportError:
        print("所需库未安装，请运行: pip install cairosvg pillow")
        return False
    except Exception as e:
        print(f"转换失败: {svg_file} - {str(e)}")
        return False

def main():
    icons_dir = "assets/icons"
    
    if not os.path.exists(icons_dir):
        print(f"目录不存在: {icons_dir}")
        return
    
    svg_files = [f for f in os.listdir(icons_dir) if f.endswith('.svg')]
    
    if not svg_files:
        print("未找到SVG文件")
        return
    
    print(f"找到 {len(svg_files)} 个SVG文件")
    
    success_count = 0
    for svg_file in svg_files:
        svg_path = os.path.join(icons_dir, svg_file)
        png_file = svg_file.replace('.svg', '.png')
        png_path = os.path.join(icons_dir, png_file)
        
        if convert_svg_to_png(svg_path, png_path):
            success_count += 1
    
    print(f"\n转换完成: {success_count}/{len(svg_files)} 个文件")

if __name__ == "__main__":
    main()

File: miniprogram/convert_svg_to_png.js
const fs = require('fs');
const path = require('path');
const { execSync } = require('child_process');

const iconsDir = path.join(__dirname, 'assets', 'icons');

function convertSvgToPng(svgFile) {
    const svgPath = path.join(iconsDir, svgFile);
    const pngFile = svgFile.replace('.svg', '.png');
    const pngPath = path.join(iconsDir, pngFile);

    try {
        execSync(`magick "${svgPath}" -resize 81x81 "${pngPath}"`, { stdio: 'inherit' });
        console.log(`✓ 转换成功: ${svgFile} -> ${pngFile}`);
        return true;
    } catch (error) {
        console.log(`✗ 转换失败: ${svgFile}`);
        console.log('提示: 请安装ImageMagick');
        return false;
    }
}

function main() {
    if (!fs.existsSync(iconsDir)) {
        console.log(`目录不存在: ${iconsDir}`);
        return;
    }

    const files = fs.readdirSync(iconsDir);
    const svgFiles = files.filter(file => file.endsWith('.svg'));

    if (svgFiles.length === 0) {
        console.log('未找到SVG文件');
        return;
    }

    console.log(`找到 ${svgFiles.length} 个SVG文件\n`);

    let successCount = 0;
    svgFiles.forEach(svgFile => {
        if (convertSvgToPng(svgFile)) {
            successCount++;
        }
    });

    console.log(`\n转换完成: ${successCount}/${svgFiles.length} 个文件`);
}

main();